lemur/tests/test_endpoints.py

```diff
@@ -353,6 +353,174 @@ def test_rotate_cli_old_to_new(session, source_plugin):
+
+    # Setup endpoints
+    ep1, ep2 = EndpointFactory(), EndpointFactory()
+
+    # Add Primary Certs to Endpoints
+    ep1.primary_certificate = primary_cert1
+    ep2.primary_certificate = primary_cert2
+
+    _setup_rotation_eligible_endpoint_sni_certificate(ep1, old_sni_cert1, new_sni_cert1)
+    _setup_rotation_eligible_endpoint_sni_certificate(ep2, old_sni_cert2, new_sni_cert2)
+
+    # Setup Source
+    source_name = "test_rotate_cli_source_sni"
+    _setup_source_for_endpoints([ep1, ep2], source_name)
+
+    session.commit()
+
+    rotate(
+        source=source_name,
+        message=None,
+        commit=True,
+        endpoint_name=None,
+        old_certificate_name=None,
+        new_certificate_name=None,
+        region=None,
+    )
+
+    # Ensure that SNI certs were rotated
+    assert ep1.sni_certificates == [new_sni_cert1]
+    assert ep2.sni_certificates == [new_sni_cert2]
+
+    # Ensure that Primary certs were no rotated
+    assert ep1.primary_certificate == primary_cert1
+    assert ep2.primary_certificate == primary_cert2
+
+
+def test_rotate_cli_by_source_multiple_sources(session, source_plugin):
+    """
+    Ensure that the CLI command 'lemur rotate_by_source --source 'test source'
+    correctly rotates all endpoints using the old certificate with the new certificate.
+    Ensure that when we have multiple Sources we are only rotating the endpoints on the
+    source that we pass in
+    """
+    from lemur.certificates.cli import rotate
+
+    # These endpoints will be associated to Source="test-source" and should be rotated
+    ep1, ep2 = EndpointFactory(), EndpointFactory()
+
+    # These endpoints will be associated to Source="test-source-other" and should NOT be rotated
+    ep3, ep4 = EndpointFactory(), EndpointFactory()
+
+    # Setup Primary Certs
+    ep1_old_cert, ep1_new_cert = CertificateFactory(), CertificateFactory()
+    ep2_old_cert, ep2_new_cert = CertificateFactory(), CertificateFactory()
+    ep3_old_cert, ep3_new_cert = CertificateFactory(), CertificateFactory()
+    ep4_old_cert, ep4_new_cert = CertificateFactory(), CertificateFactory()
+
+    # Setup all endpoints to be eligible for rotation
+    _setup_rotation_eligible_endpoint_primary_certificate(
+        endpoint=ep1, old_primary_certificate=ep1_old_cert, new_primary_certificate=ep1_new_cert
+    )
+    _setup_rotation_eligible_endpoint_primary_certificate(
+        endpoint=ep2, old_primary_certificate=ep2_old_cert, new_primary_certificate=ep2_new_cert
+    )
+    _setup_rotation_eligible_endpoint_primary_certificate(
+        endpoint=ep3, old_primary_certificate=ep3_old_cert, new_primary_certificate=ep3_new_cert
+    )
+    _setup_rotation_eligible_endpoint_primary_certificate(
+        endpoint=ep4, old_primary_certificate=ep4_old_cert, new_primary_certificate=ep4_new_cert
+    )
+
+    # Associated ep1 and ep2 with Source.label="test-source"
+    source_name = "rotate_cli_source_multiple"
```

Comment:
```suggestion
    source_name = "test-source"
```

Reply:
I updated the tests to cleanup(delete) sources on tests. see this commit: https://github.com/DataDog/lemur/pull/112/commits/0687beab47bb5fa9ce4ab263e2069b6618d13c02

For context, the reason for matching the source name to the test name is because the source label needs to be a unique identifier in the DB. Deleting the sources after the test allows us to reuse source names and makes the tests easier to follow IMO.